src/coprocessor/dag/expr.rs

```diff
@@ -0,0 +1,238 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::io;
+use std::convert::{TryFrom, TryInto};
+
+use super::super::codec::mysql::{Duration, Time, Decimal};
+use super::super::codec::Datum;
+use super::super::super::util;
+
+use chrono::FixedOffset;
+
+use tipb::expression::{ExprType, ScalarFuncSig};
+use tipb::expression::Expr;
+use tipb::select::DAGRequest;
+
+use util::codec::number::{NumberDecoder, NumberEncoder};
+use coprocessor::codec::mysql::decimal::DecimalDecoder;
+use coprocessor::codec::mysql::MAX_FSP;
```

Comment:
Module `copressor` and `super::super` are the same module. Should we put them together?